### PR TITLE
S2583/S2589: Add FP repro for #8495

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Reproduces #8495

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs [hunk 1]
```diff
@@ -3691,3 +3691,109 @@ void TestMethod()
...
+                return true;
+            }
+            if (b)
+            {
+                flag = false;
+            }
+        }
+        finally
+        {
+            a.ToString();
```
Comment (cristian-ambrosini-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs:3764]: Educational: what is the purpose of this invocation?

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs:3764]: To make sure that the `finally` will not get optimized away.

Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs:3764]: It's a good question!

**Pull request merged**